core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt

```diff
@@ -62,15 +35,51 @@ class Primitive<T : ValidType> : TypeExpression<T> {
     }
 }
 
-fun String.toStringType(): Primitive<StringType> = Primitive(this)
+class NumberPrimitive(private val value: Number) : Primitive<NumberType>() {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = "$value",
+        parameters = emptyMap(),
+    )
+}
+
+class StringPrimitive(private val value: String) : Primitive<StringType>() {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = "\"$value\"",
+        parameters = emptyMap(),
+    )
+}
+
+class BooleanPrimitive(private val value: Boolean) : Primitive<BooleanType>() {
+    override fun toDopeQuery() = DopeQuery(
+        queryString = when (value) {
+            true -> TRUE.toDopeQuery().queryString
+            false -> FALSE.toDopeQuery().queryString
+        },
+        parameters = emptyMap(),
+    )
+}
+
+class ArrayPrimitive<T : ValidType>(private val collection: Collection<TypeExpression<out T>>) : Primitive<ArrayType<T>>() {
```

Comment:
if you use my approach above you can use with the keyword let ensure that toDopeQuery is only called once:
`... : Primitive<ArrayType<T>>(collection.map { it.toDopeQuery() }.let {...`